leaderboard/redis/cluster_client.go

```diff
@@ -0,0 +1,208 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	goredis "github.com/go-redis/redis/v8"
+)
+
+type clusterClient struct {
+	*goredis.ClusterClient
+}
+
+type ClusterOptions struct {
+	Hosts    []string
+	Password string
+}
+
+// NewClusterClient returns a new redis instance
+func NewClusterClient(clusterOptions ClusterOptions) *clusterClient {
+	goRedisClient := goredis.NewClusterClient(&goredis.ClusterOptions{
+		Addrs:    clusterOptions.Hosts,
+		Password: clusterOptions.Password,
+	})
+
+	return &clusterClient{goRedisClient}
+}
+
+// ExpireAt call redis EXPIREAT function
+func (cc *clusterClient) ExpireAt(ctx context.Context, key string, time time.Time) error {
+	result, err := cc.ClusterClient.ExpireAt(ctx, key, time).Result()
+	if err != nil {
+		return err
+	}
+
+	if result != true {
+		return NewKeyNotFoundError(key)
+	}
+
+	return nil
+}
+
+// Ping call redis PING function
+func (cc *clusterClient) Ping(ctx context.Context) error {
+	err := cc.ClusterClient.Ping(ctx).Err()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// SAdd call redis SADD function
+func (cc *clusterClient) SAdd(ctx context.Context, key, member string) error {
+	_, err := cc.ClusterClient.SAdd(ctx, key, member).Result()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// SRem call redis SREM function
+func (cc *clusterClient) SRem(ctx context.Context, key, member string) error {
+	err := cc.ClusterClient.SRem(ctx, key, member).Err()
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
You are not doing anything with `err`. You can replace this with:
```suggestion
	return cc.ClusterClient.SRem(ctx, key, member).Err()	
```